code/game/machinery/autolathe/nanoforge.dm

```diff
@@ -96,5 +96,5 @@
 /obj/machinery/autolathe/nanoforge/check_user(mob/user)
 	if(user.stats.getPerk(PERK_TECHNOMANCER) || user.stat_check(STAT_MEC, STAT_LEVEL_EXPERT))
```

Comment:
You could add a check for user.stats in case a mob without stats tries to use the nanoforge.

`if(user.stats && (user.stats.getPerk(PERK_TECHNOMANCER) || user.stat_check(STAT_MEC, STAT_LEVEL_EXPERT)))`

Reply:
> You could add a check for user.stats in case a mob without stats tries to use the nanoforge.
> 
> `if(user.stats && (user.stats.getPerk(PERK_TECHNOMANCER) || user.stat_check(STAT_MEC, STAT_LEVEL_EXPERT)))`

All mobs should have stats, except when the mob is destroyed. A runtime given because "stats" is null should give us an indication that there is something wrong with the code.

Reply:
> You could add a check for user.stats in case a mob without stats tries to use the nanoforge.
> 
> `if(user.stats && (user.stats.getPerk(PERK_TECHNOMANCER) || user.stat_check(STAT_MEC, STAT_LEVEL_EXPERT)))`

You'd probably wanna use `if(user?.stats.getPerk(PERK_TECHNOMANCER) || user.stat_check(STAT_MEC, STAT_LEVEL_EXPERT)))` instead right? Last I checked that worked fine.